### PR TITLE
Formating Python datetime object to ISO when outputting a GeoJSON

### LINKED ISSUE #13: CLI: Format Python datetime object to be human readable when outputting a GeoJSON
Check how the API does this.

### PR DESCRIPTION
**Pull request by hn437**
### Description
The datetime object in the results of indicators prevent the result from being dumped as geojson. When dumping, the object will be converted into an iso format like FastAPI does it for the api return

### Corresponding issue
Closes #13 

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code

### CODE REVIEW

File: workers/ohsome_quality_analyst/cli.py [hunk 1]
```diff
@@ -37,6 +38,12 @@ def _add_opts(func):
     return _add_opts
 
 
+def format_datetime_for_dump(o):
+    """Checks for datetime objects and converts them to ISO 8601 format"""
+    if isinstance(o, (datetime.date, datetime.datetime)):
+        return o.isoformat()
```
Comment (matthiasschaub) [inline workers/ohsome_quality_analyst/cli.py:44]:
```suggestion
def to_iso_timestring(time: datetime) -> str:
    """Checks for datetime objects and converts them to ISO 8601 format"""
    if isinstance(time, (datetime.date, datetime.datetime)):
        return time.isoformat()
```

Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/cli.py:44]: Just a little naming suggestion.

**Comment by matthiasschaub** [comment]
@hn437 That`s a nice solution!

I left a minor comment on naming.

It would be nice to have a test covering that function as well.

**Comment by Hagellach37** [comment]
Hey @hn437 ,
as a general comment I'm not sure if this is the right place for the function. It might be better place in [utils](https://github.com/GIScience/ohsome-quality-analyst/tree/main/workers/ohsome_quality_analyst/utils).

similar one as from @matthiasschaub : also you could rename the function to be more precise: .e.g. `datetime_to_isostring_timestamp` or similar.

And then I have a question on if this function actually works as expected. In the python docs it says:

"default should be a function that gets called for objects that can’t otherwise be serialized. It should return a JSON encodable version of the object or raise a TypeError. If not specified, TypeError is raised."

I think a simple test for this would be good. :)

Other than that, good work.

**Comment by hn437** [comment]
Thank you for the suggestions. I changed the variable naming and definition as @matthiasschaub suggested and renamed it as well as moved it to the helper script in [utils](https://github.com/GIScience/ohsome-quality-analyst/tree/main/workers/ohsome_quality_analyst/utils) as @Hagellach37 suggested.

By now I just tested it locally and it worked, it makes the dump possible and does not corrupt anything else. But yes, a test should be added and was planned for the geojson creation in general.
However, I've never created a test. Could someone maybe tell me in short how this is generally done and what to keep in mind? Maybe in a short Meeting on Wednesday?

**Comment by matthiasschaub** [comment]
> Thank you for the suggestions. I changed the variable naming and definition as @matthiasschaub suggested and renamed it as well as moved it to the helper script in [utils](https://github.com/GIScience/ohsome-quality-analyst/tree/main/workers/ohsome_quality_analyst/utils) as @Hagellach37 suggested.
> 
> By now I just tested it locally and it worked, it makes the dump possible and does not corrupt anything else. But yes, a test should be added and was planned for the geojson creation in general.
> However, I've never created a test. Could someone maybe tell me in short how this is generally done and what to keep in mind? Maybe in a short Meeting on Wednesday?

I am happy to jump on a call with you to talk you through our test setup.

**Review by matthiasschaub — APPROVED** [review]

**Review by matthiasschaub — APPROVED** [review]

**Review by matthiasschaub — APPROVED** [review]

**Pull request merged**